src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -146,59 +168,75 @@ internal static IEnumerable<string> GetDefaultAvailableModuleFiles(bool force, b
             }
         }
 
-        internal static List<string> GetModuleVersionsFromAbsolutePath(string directory)
+        /// <summary>
+        /// Get a list of module files from the given directory without recursively searching all sub-directories.
+        /// This method assumes the given directory is a module folder or a version sub-directory of a module folder.
+        /// </summary>
+        internal static List<string> GetModuleFilesFromAbsolutePath(string directory)
         {
             List<string> result = new List<string>();
             string fileName = Path.GetFileName(directory);
-            Version moduleVersion;
-            // if the user give the module path including version, we should be able to find the module as well
-            if (Version.TryParse(fileName, out moduleVersion) && Directory.Exists(Directory.GetParent(directory).ToString()))
-            {
-                fileName = Directory.GetParent(directory).Name;
-            }
-            foreach (var version in GetModuleVersionSubfolders(directory))
+
+            // If the given directory doesn't exist or it's the root folder, then return an empty list.
+            if (!Directory.Exists(directory) || fileName == string.Empty) { return result; }
+
+            // If the user give the module path including version, the module name could be the parent folder name.
+            if (Version.TryParse(fileName, out Version ver))
             {
-                var qualifiedPathWithVersion = Path.Combine(directory, Path.Combine(version.ToString(), fileName));
-                string manifestPath = qualifiedPathWithVersion + StringLiterals.PowerShellDataFileExtension;
-                if (File.Exists(manifestPath))
-                {
-                    bool isValidModuleVersion = version.Equals(ModuleIntrinsics.GetManifestModuleVersion(manifestPath));
+                string parentDirPath = Path.GetDirectoryName(directory);
```

Comment:
I assume rather than throw this returns `null` or `""`?

Reply:
The method is removed now.

Reply:
This method returns an empty string. I changed to use `string.IsNullOrEmpty` below.